src/main/java/com/yelp/nrtsearch/server/luceneserver/DeleteIndexBackupHandler.java

```diff
@@ -0,0 +1,104 @@
+/*
+ * Copyright 2020 Yelp Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.yelp.nrtsearch.server.luceneserver;
+
+import com.yelp.nrtsearch.server.grpc.DeleteIndexBackupRequest;
+import com.yelp.nrtsearch.server.grpc.DeleteIndexBackupResponse;
+import com.yelp.nrtsearch.server.utils.Archiver;
+import com.yelp.nrtsearch.server.utils.VersionedResourceObject;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.Calendar;
+import java.util.Date;
+import java.util.List;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class DeleteIndexBackupHandler
```

Comment:
You can add a test for this class as well. It would involve creating a test object of this class by passing a mocked `Archiver` object and then testing the `handle` method (and optionally `olderThanNDays` - but you should be able to test it by testing just the handle method)

Reply:
I added the test of `DeleteIndexBackupHandler` with mocked `Archiver`. In that test I only verify that a single one old resource is deleted, appropriate methods are called and the correct response is returned. Also I have a separate test for `isOlderThanNDays` function. 

In order to make a better test of handler I would need to mock the return value of `Instant.now()` called inside the handler. Basing on `Instant.now()` the handler decides which resources are older than `n` days. As far as I know it's possible to mock `Instant.now()` with `PowerMock` library. Are you ok with adding this to the dependency list for this project? Or do you know better ways of mocking system time with Mockito (I couldn't find a solution with that lib)?

Reply:
Instead of mocking system clock, the tests set the creation timestamps of object in a following fashion 
```Instant date1 = Instant.now().minus(20, ChronoUnit.DAYS);```